src/test/java/org/cactoos/scalar/AvgOfTest.java

```diff
@@ -51,125 +51,308 @@ public void withEmptyCollection() {
+            Matchers.equalTo(100.2665f)
         );
     }
 
     @Test
-    public void withFloatCollection() {
+    public void withFloatCollectionIntValue() {
         MatcherAssert.assertThat(
             new AvgOf(
-                new ListOf<>(1.0f, 2.0f, 3.0f, 4.0f).toArray(new Float[4])
+                1.0f, 2.0f, 3.0f, 4.0f
             ).intValue(),
             Matchers.equalTo(2)
         );
+    }
+
+    @Test
+    public void withFloatCollectionLongValue() {
         MatcherAssert.assertThat(
             new AvgOf(
-                new ListOf<>(1.0f, 2.0f, 3.0f, 4.0f).toArray(new Float[4])
+                1.0f, 2.0f, 3.0f, 4.0f
             ).longValue(),
             Matchers.equalTo(2L)
         );
+    }
+
+    @Test
+    public void withFloatCollectionDoubleValue() {
         MatcherAssert.assertThat(
             new AvgOf(
-                new ListOf<>(1.0f, 2.0f, 3.0f, 4.0f).toArray(new Float[4])
+                1.0f, 2.0f, 3.0f, 4.0f
             ).doubleValue(),
             Matchers.equalTo(2.5d)
         );
+    }
+
+    @Test
+    public void withFloatCollectionFloatValue() {
         MatcherAssert.assertThat(
             new AvgOf(
-                new ListOf<>(1.0f, 2.0f, 3.0f, 4.0f).toArray(new Float[4])
+                1.0f, 2.0f, 3.0f, 4.0f
             ).floatValue(),
             Matchers.equalTo(2.5f)
         );
     }
 
     @Test
-    public void withScalars() {
+    public void withFloatCollectionMaxValue() {
+        MatcherAssert.assertThat(
+            new AvgOf(
+                Float.MAX_VALUE, Float.MAX_VALUE
+            ).floatValue(),
+            Matchers.equalTo(Float.MAX_VALUE)
+        );
+    }
+
+    @Test
+    public void withFloatCollectionMinValue() {
+        MatcherAssert.assertThat(
+            new AvgOf(
+                Float.MIN_VALUE, Float.MIN_VALUE
+            ).floatValue(),
+            Matchers.equalTo(Float.MIN_VALUE)
+        );
+    }
+
+    @Test
+    public void withIntScalarsIntValue() {
+        MatcherAssert.assertThat(
+            new AvgOf(
+                () -> 1, () -> 2, () -> 10
+            ).intValue(),
+            Matchers.equalTo(4)
+        );
+    }
+
+    @Test
+    public void withLongScalarsIntValue() {
         MatcherAssert.assertThat(
             new AvgOf(
                 () -> 1L, () -> 2L, () -> 10L
+            ).intValue(),
+            Matchers.equalTo(4)
+        );
+    }
+
+    @Test
+    public void withFloatScalarsIntValue() {
+        MatcherAssert.assertThat(
+            new AvgOf(
+                () -> 1f, () -> 2f, () -> 10f
             ).longValue(),
             Matchers.equalTo(4L)
         );
+    }
+
+    @Test
+    public void withDoubleScalarsIntValue() {
+        MatcherAssert.assertThat(
+            new AvgOf(
+                () -> 1.0d, () -> 2.0d, () -> 10.0d
+            ).intValue(),
+            Matchers.equalTo(4)
+        );
+    }
+
+    @Test
+    public void withIterableOfScalars() {
```

Comment:
in this test type of `new IterableOf<Scalar<Number>>(() -> 1L, () -> 2L, () -> 10L)` can be inferred, so params are not neccecary

Reply:
Without it the compiler complains:

```
[WARNING] yegor256/cactoos/src/test/java/org/cactoos/scalar/AvgOfTest.java:[358,17] unchecked generic array creation for varargs parameter of type org.cactoos.Scalar<java.lang.Number>[]
```

And we have `-Werror` enabled.